### DISCUSSION TITLE
v2024.02.2_26.0_x86-64_0.37

### DISCUSSION THREAD

**Discussion by PartialVolume** [PartialVolume/shredos.x86_64#258] [Announcements]
v2024.02.2_26.0_x86-64_0.37

| Buildroot Version | Kernel Version | Shredos Minor version | Architecture | Nwipe Version |
| --- | --- | --- | --- | --- |
| 2024.02.2|6.6.22|26.0|x86-64|0.37|

Other software versions used within ShredOS for disc related operations
| hexedit | hdparm | Smartctl | Sg3_Utils   | nvme-cli |
|--- |--- |----| --- | --- |
| 1.6 | 9.65 | 7.3 |1.47|1.12|

## Notable changes in this release - ShredOS
- Updated buildroot from 23.08.2 to 24.02.2, many changes to the Linux kernel and other software, over 4000 files updated!
- Updated nwipe from v0.35 to v0.37
- Added new ShredOS option: `shredos_config="..."`. This can be added to the kernel command line that allow the user to obtain nwipe.conf and nwipe_customers.csv from a tftp or ftp server on the local network as ShredOS boots up.
- Added a new ShredOS option: `shredos_output="..."`. This allows the user to configure ShredOS to automatically send PDF erasure certificates and logs to a local tftp or ftp server. The older method `lftp=` outputs PDF certificates to a ftp server but required the user to specify ftp commands and therefore a knowledge of either ftp or tftp commands. The `lftp=` option is deprecated and may be removed in a future version. This new method only requires details of the servers IP, path, username and password, to form part of the shredos_output command and therefore no knowledge of tftp/ftp commands are required other than to setup a server.
- Added a new ShredOS option: `shredos_autoreboot=enable`. Placing this option on the kernel command line will automatically reboot the computer after Nwipe has completed the erase and ShredOS has written the certificates, logs and configs to the local tftp/ftp server (if user has specified tftp/ftp transfers). If PDF certificates are being output to tftp/ftp servers and the transfer should fail for whatever reason, ShredOS will pause the reboot to allow the user to investigate the network problem. This is useful when you want to automate the installation of a operating system if booting ShredOS via the network using netboot ,onto the erased discs after a erasure has been completed. See #225
- Added a new ShredOS option: `shredos_exclude_boot_disc=yes`. This will exclude the FAT formatted USB boot drive, as created by dd/Rufus or Ventoy, that contains ShredOS, from being displayed in the drive selection screen when in GUI mode and excluded from the drives that will get wiped, when in autonuke  mode (autonuke enabled by --autonuke in the nwipe_options=".." on the kernel command line). Alternatively, there is a second method you can use to exclude the boot disc. Create a empty file on the ShredOS boot disk at this specific location `/etc/shredos/shredos_exclude_disc`. See [How to exclude the FAT formatted ShredOS Boot drive from Nwipe, interactive and autonuke modes](https://github.com/PartialVolume/shredos.x86_64?tab=readme-ov-file#how-to-exclude-the-fat-formatted-shredos-boot-drive-from-nwipe-interactive-and-autonuke-modes) for further details.
- When ShredOS has been configured to retrieve config files from a tftp/ftp server, it will now wait for up to 30 seconds for the network interface and DHCP request to be answered before launching nwipe. This wait will only occur if the user specified either shredos_config="..." or shredos_ouput="..." on the kernel command line. This fixes some systems where the network had not yet been fully initialised and IP address obtained before nwipe was launched. This caused nwipe to be unable to retrieve config files when the user had configured ShredOS to retrieve those files from a tftp/ftp server. Now ShredOS will expect the server to respond to a ping before it will launch nwipe. If no ping within 30 seconds nwipe will be launched anyway and will log a warning regarding the server not reponding to the /transfer.log.
- On exit from nwipe, ShredOS will now provide a archive to USB option if for any reason it was unable to save the PDF documents and logs to either USB or to a network server if a network server has been configured using the shredos_output kernel command line option.
- Increased ShredOS's vfat partition size from 250MB to 280MB.

## Notable changes in this release - Nwipe v0.37
- Added the XORoshiro-256 pseudo random number generator (PRNG). Thanks to Fabian Druschke @knogle [#555](https://github.com/martijnvanbrummelen/nwipe/pull/555)
- Added  the Lagged Fibonacci PRNG generator. Thanks again to Fabian Druschke @knogle #556
- Changed the default method from "DOD Short" to "prng stream", using the XORoshiro-256 prng
- Added missing help for HMG IS5 enhanced. Thanks to @AndCycle #569
- Fixed an issue in configure.ac which was producing an error while running `./configure`,  mentioning libconfig, however the presence of libconfig had already been checked for, earlier in configure.ac. Although this error did not cause `./configure` to abort prematurely and therefore make would build the source correctly, it did cause a issue for inclusion into Debian Sid. Thanks to @Polynomial-C #574
- Minor change to nwipe's man page filename, nwipe.1 to nwipe.8 to fix a Debian warning. #577

## Download Counters
| Architecture/format/mode | Download link |
| ------------------------- | --------------------|
|  x86-64 .img for USB Vanilla | [![GitHub release (by file)](https://img.shields.io/github/downloads/PartialVolume/shredos.x86_64/v2024.02.2_26.0_x86-64_0.37/shredos-2024.02.2_26.0_x86-64_0.37_20240610.img?label=%20)](https://github.com/PartialVolume/shredos.x86_64/releases/download/v2024.02.2_26.0_x86-64_0.37/shredos-2024.02.2_26.0_x86-64_0.37_20240610.img) |
|  x86-64 .iso for CD/DVD/Ventoy nomodeset | [![GitHub release (by file)](https://img.shields.io/github/downloads/PartialVolume/shredos.x86_64/v2024.02.2_26.0_x86-64_0.37/shredos-2024.02.2_26.0_x86-64_0.37-nomodeset_20240611.iso?label=%20)](https://github.com/PartialVolume/shredos.x86_64/releases/download/v2024.02.2_26.0_x86-64_0.37/shredos-2024.02.2_26.0_x86-64_0.37-nomodeset_20240611.iso) |
|  x86-64 .iso for CD/DVD/Ventoy Vanilla | [![GitHub release (by file)](https://img.shields.io/github/downloads/PartialVolume/shredos.x86_64/v2024.02.2_26.0_x86-64_0.37/shredos-2024.02.2_26.0_x86-64_0.37_20240611.iso?label=%20)](https://github.com/PartialVolume/shredos.x86_64/releases/download/v2024.02.2_26.0_x86-64_0.37/shredos-2024.02.2_26.0_x86-64_0.37_20240611.iso) |

## .iso and .img Notes
- The .img file for writing to a USB flash drive is both legacy (bios) and UEFI bootable.
- The .iso file for CD-R/DVD-R is legacy bootable only if burned to DVD/CD, however if you copy the .iso to a Ventoy USB stick it is bootable on both UEFI & legacy hardware.

## Ventoy Notes
If Ventoy's 'normal' mode does not work try 'Ventoy's 'grub2' mode.  To enable grub2 mode in Ventoy, when the Ventoy list of Iso/img files appear type control +r to toggle into grub2 mode, then select the ShredOS .iso. If you try to boot the ShredOS .iso using Ventoy's 'normal' mode you will end up at the grub prompt on some systems.
 
**What's the difference between Ventoy's normal & grub2 modes?**
Well it appears that in normal mode Ventoy uses ShredOS's own grub code to boot ShredOS, while in grub2 mode Ventoy uses it's own code. Using Ventoys own grub code when in grub2 mode may allow you to boot when normal mode doesn't work however, Ventoy's grub does not load the progress module so you won't see the progress percentage as the ShredOS kernel is loaded. ShredOS also includes modules for PXE booting.

The ShredOS .iso labeled nomodeset disables the Linux Direct Rendering Manager (DRM) and so remains in simple framebuffer mode. This mode is compatible with most graphics hardware and avoids any buggy graphics drivers. This would be used when you find that the standard version of ShredOS initially displays text but then fails to display nwipe.

## 64bit .img and .iso for modern Intel & compatible processors post Pentium 4.
The .iso is for burning to DVD/CD and copying onto a Ventoy USB stick.

## 32bit .img and .iso for Pentium 4 and earlier CPUs.
Pentium MMX, K5, K6, K6-II, K6-III, Pentium Pro, Pentium II, Pentium III, Athlon, Athlon XP, Opteron*, Athlon FX-51*, Athlon FX-53*, Athlon 64* Pentium 4 (32 bit).

. * = 32-bit/64-bit. The x86 32 bit version of ShredOS will often run on x86 64 processors but not all 64 bit processors due to limitations in the CPU or bios.

## Checksums
```
sha1 1eeefec718f54911355c07db09bc10863f1c63cf  shredos-2024.02.2_26.0_x86-64_0.37_20240610.img
sha1 375c7667b683184cbb5577126be980ca1c3d1d65  shredos-2024.02.2_26.0_x86-64_0.37_20240611.iso
sha1 d210b9974eb295e1644ca5de488cd0b01750fc6b  shredos-2024.02.2_26.0_x86-64_0.37-nomodeset_20240611.iso
```

<hr /><em>This discussion was created from the release <a href='https://github.com/PartialVolume/shredos.x86_64/releases/tag/v2024.02.2_26.0_x86-64_0.37'>v2024.02.2_26.0_x86-64_0.37</a>.</em>

**Comment by Nebuli1** [comment]
wow, cool, I'm going to open the wine and drink healthily. I waited and kept my fingers crossed, it's a pity I wasn't of much use essential. Thank you very much for all your work.

**Comment by PartialVolume** [comment]
I'll be adding some notes to the [README.md](https://github.com/PartialVolume/shredos.x86_64/blob/master/README.md)  regarding the use of shredos_output="..." and shredos_config="...".

I'll also be adding the i586 (32bit) builds.

**Comment by tadvaas** [comment]
Is it possible to retrieve shredos_config using http protocol?

Such as `shredos_config="http:192.168.0.26:/shredos/etc/nwipe/::no password:d`? I run shredOS via iPXE server and for the love of God I cannot get nwipe.conf to work. Is it only possible to load this via ftp or sftp? However, I tried that as well, however, no luck.

**Reply by PartialVolume** [reply]
No, there is no http option to retrieve nwipe.conf, however, there is a option to send the reports via http but not to retrieve the files.

I can offer some suggestions in regards fixing your FTP, TFTP or sftp if you can post some more info about your server setup.

Can you post the contents of the file transfer.log? 

The command you are using `shredos_config="http:192.168.0.26:/shredos/etc/nwipe/::no password:d` has the 'd' right at the end which enables verbose debug info to be sent to transfer.log. So that file should show you what the problem is.

**Reply by tadvaas** [reply]
Thanks @PartialVolume 

I managed to fix it by simply using FTP protocol for shredos_config and shredos_output (least amount of hassle):

iPXE boot menu file:

`:shredos
dhcp
kernel ${base-url}/shredos/boot/shredos console=tty3 quiet loglevel=0 loadkeys=uk rw \
        nwipe_options="--method=zero --verify=off --noblank --autonuke --autopoweroff --logfile=/dev/null" \
		shredos_config="ftp:192.168.0.18:/home/shredos/config/:***:***:" \
		shredos_output="ftp:192.168.0.18:/home/shredos/reports/:***:***:"
boot`

However, we do not need nwipe log file to be transferred due to large amount of devices being sanitised unattended, shredos_output tries to find nwipe_log*.txt file, and when it does not find it it breaks the autopoweroff. Any way to overcome that?

Also 

>  there is a option to send the reports via http but not to retrieve the files.

That's interesting, tell me more.
